### PR TITLE
Cloud-684 slack notifications

### PR DESCRIPTION
**Pull request by nmarukovich**
[CLOUD-684](https://jira.percona.com/browse/CLOUD-684) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/recreate.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/storage.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/haproxy.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/scheduled-backup.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/big-data.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/upgrade-haproxy.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/self-healing-chaos.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/init-deploy.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/restore-to-encrypted-cluster.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/security-context.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/self-healing-advanced.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/self-healing-advanced-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/upgrade-proxysql.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/limits.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/demand-backup.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/monitoring-2-0.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/operator-self-healing-chaos.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/smart-update.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/proxysql-sidecar-res-limits.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/upgrade-consistency.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/users.log) |
| pitr | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/pitr.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/tls-issue-self.log) |
| tls-issue-cert-manager | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/dbfef335/tls-issue-cert-manager.log) |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1109/commits/dbfef335c7c883431d28a1225b0dc342c90972c2
image: `perconalab/percona-xtradb-cluster-operator:PR-1109-dbfef335`

**Review by tplavcic — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,7 @@ require (
 	github.com/robfig/cron/v3 v3.0.1
 	go.uber.org/zap v1.18.1
 	k8s.io/api v0.19.2
-	k8s.io/apimachinery v0.23.3
+	k8s.io/apimachinery v0.23.4
```
Comment (nmarukovich) [inline go.mod:26]: I can't build the image without this changes.

Comment (egegunes) [reply go.mod:26]: what is the error?

File: Jenkinsfile [hunk 2]
```diff
@@ -376,7 +376,13 @@ pipeline {
             script {
                 setTestsresults()
                 if (currentBuild.result != null && currentBuild.result != 'SUCCESS') {
```
Comment (egegunes) [inline Jenkinsfile:378]: maybe it's better to send notifications on success too?

Comment (nmarukovich) [reply Jenkinsfile:378]: I think yes, but maybe in next iteration))

File: go.mod (hunk 1 above)
Comment (nmarukovich) [reply go.mod:26]: #13 [go_builder 4/5] RUN mkdir -p build/_output/bin     && GOOS=linux GOARCH=amd64 CGO_ENABLED=0 GO_LDFLAGS=-w -s -trimpath         go build -mod=vendor -ldflags "-w -s -X main.GitCommit=7f2d8575a3fc8de00019bdbb090bc543831b7ae9 -X main.GitBranch=main -X main.BuildTime=2022-03-01T19:45:06Z"            -o build/_output/bin/percona-xtradb-cluster-operator                cmd/manager/main.go     && cp -r build/_output/bin/percona-xtradb-cluster-operator /usr/local/bin/percona-xtradb-cluster-operator
#13 sha256:b76bf0a19f58ca1b3897534a507a9d75442cd0e20bc9b9eeb2d06e7200cabdcc
#13 DONE 179.7s

#14 [go_builder 5/5] RUN go get k8s.io/apimachinery/pkg/util/sets     && go build -o build/_output/bin/peer-list cmd/peer-list/main.go     && cp -r build/_output/bin/peer-list /usr/local/bin/
#14 sha256:dc56837ad08e7068233a280954077dd9096fed99d16e7c594c4a389d470e24db
#14 14.01 go: downloading k8s.io/apimachinery v0.19.2
#14 23.41 go get: upgraded k8s.io/apimachinery v0.23.3 => v0.23.4
#14 23.52 go: inconsistent vendoring in /go/src/github.com/percona/percona-xtradb-cluster-operator:
#14 23.52 	k8s.io/apimachinery@v0.23.4: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#14 23.52 	k8s.io/apimachinery@v0.23.3: is marked as explicit in vendor/modules.txt, but not explicitly required in go.mod
#14 23.52
#14 23.52 	To ignore the vendor directory, use -mod=readonly or -mod=mod.
#14 23.52 	To sync the vendor directory, run:
#14 23.52 		go mod vendor
#14 ERROR: executor failed running [/bin/sh -c go get k8s.io/apimachinery/pkg/util/sets     && go build -o build/_output/bin/peer-list cmd/peer-list/main.go     && cp -r build/_output/bin/peer-list /usr/local/bin/]: exit code: 1
------
 > [go_builder 5/5] RUN go get k8s.io/apimachinery/pkg/util/sets     && go build -o build/_output/bin/peer-list cmd/peer-list/main.go     && cp -r build/_output/bin/peer-list /usr/local/bin/:
------
executor failed running [/bin/sh -c go get k8s.io/apimachinery/pkg/util/sets     && go build -o build/_output/bin/peer-list cmd/peer-list/main.go     && cp -r build/_output/bin/peer-list /usr/local/bin/]: exit code: 1
You can see the same error in main branch

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/haproxy.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/storage.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/upgrade-haproxy.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/big-data.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/scheduled-backup.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/recreate.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing.log) |
| self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing-chaos.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/init-deploy.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/restore-to-encrypted-cluster.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/security-context.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/upgrade-proxysql.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/limits.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing-advanced.log) |
| self-healing-advanced-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing-advanced-chaos.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/demand-backup.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/monitoring-2-0.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/operator-self-healing.log) |
| operator-self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/operator-self-healing-chaos.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/smart-update.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/auto-tuning.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/pitr.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/users.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/upgrade-consistency.log) |
| demand-backup-encrypted-with-tls | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/demand-backup-encrypted-with-tls.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/tls-issue-self.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/tls-issue-cert-manager-ref.log) |
| validation-hook | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/validation-hook.log) |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1109/commits/e2106f3585cc80bdc62f09dff40024c59efc8705
image: `perconalab/percona-xtradb-cluster-operator:PR-1109-e2106f35`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/scheduled-backup.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/haproxy.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/upgrade-haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/limits.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/monitoring-2-0.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/recreate.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/big-data.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/affinity.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/scaling-proxysql.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/restore-to-encrypted-cluster.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/one-pod.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/demand-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/security-context.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/auto-tuning.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/pitr.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/proxysql-sidecar-res-limits.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/demand-backup-encrypted-with-tls.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/users.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/tls-issue-self.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/tls-issue-cert-manager-ref.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/validation-hook.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/storage.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing-chaos.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing-advanced.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/self-healing-advanced-chaos.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/operator-self-healing-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/upgrade-proxysql.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/smart-update.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1109/e2106f35/upgrade-consistency.log) |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1109/commits/e2106f3585cc80bdc62f09dff40024c59efc8705
image: `perconalab/percona-xtradb-cluster-operator:PR-1109-e2106f35`

**Review by tplavcic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**